app/views/render_async/_lazy_load.js.erb

```diff
@@ -0,0 +1,21 @@
+var lazyLoaded = function(element) {
+    element.setAttribute("data-loaded", true)
+}
+
+var isLazyLoaded = function (element) {
+    return "true" === element.getAttribute("data-loaded")
+};
+var _lazyListener = function () {
+    var observer = new IntersectionObserver(function (entries, observer) {
+        entries.forEach(function (entry) {
+            if(0 < entry.intersectionRatio || entry.isIntersecting){
+                if(!isLazyLoaded(entry.target)){
+                    _listener();
+                    observer.unobserve(entry.target);
+                }
+            }
+        })
+    }, {rootMargin: '0px', threshold: 0});
```

Comment:
Maybe we could make this configurable!

Reply:
Sure, will this works?

`render_async partial_path, lazy_load: {root_margin: '10px', threshold:  "Array OR single integer value"}`

Reply:
Yep, something like that will be great!